### PR TITLE
common/json: json parser sets val data even if entity is array or struct

### PR DESCRIPTION
**Pull request by yehudasa**
json parser sets val data even if entity is array (or struct).

Fixes: http://tracker.ceph.com/issues/37706
Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/test/common/test_json_formatter.cc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+   JSONParser parser;
+   ASSERT_TRUE(parser.parse(outstring.c_str(), outstring.size()));
+
+   vector<string> v;
+
+   ASSERT_TRUE (!parser.is_array());
```
Comment (tchaikov) [inline src/test/common/test_json_formatter.cc]: might want to remove the space after `ASSERT_TRUE`

File: src/test/common/CMakeLists.txt [hunk 2]
```diff
@@ -73,6 +73,14 @@ add_ceph_unittest(unittest_json_formattable ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/un
 # add_dependencies(unittest_json_formattable ceph-common)
 target_link_libraries(unittest_json_formattable ceph-common global ${BLKID_LIBRARIES})
 
+# unittest_json_formatter
+add_executable(unittest_json_formatter
+  test_json_formatter.cc
+  )
+add_ceph_unittest(unittest_json_formatter ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/unittest_json_formatter)
+# add_dependencies(unittest_json_formatter ceph-common)
+target_link_libraries(unittest_json_formatter ceph-common global ${BLKID_LIBRARIES})
```
Comment (tchaikov) [inline src/test/common/CMakeLists.txt]: i think there is no need to link against `global` or `${BLKID_LIBRARIES}` here.

**Comment by dzafman** [comment]
This fixes the `rados list-inconsistent-pg` breakage.

**Comment by tchaikov** [comment]
this also fixes "Command crashed: "sudo TESTDIR=/home/ubuntu/cephtest bash -c 'ceph_test_admin_socket_output --all', see http://pulpito.ceph.com/kchai-2018-12-20_12:03:44-rados-wip-kefu2-testing-2018-12-20-1729-distro-basic-smithi/3383910/

File: src/test/common/CMakeLists.txt (hunk 2 above)
Comment (liewegas) [reply src/test/common/CMakeLists.txt]: let's fix this in a follow-up, need to get this merged

**Pull request merged**